modules/custom-status/custom-status.php

```diff
@@ -583,24 +596,23 @@ public static function update_custom_status( int $status_id, array $args = [] ):
 		// Reset our internal object cache
 		self::$custom_statuses_cache = [];
 
-		// Prevent user from changing draft name or slug
-		if ( self::is_restricted_status( $old_status->slug )
-		&& (
-			( isset( $args['name'] ) && $args['name'] !== $old_status->name )
-			||
-			( isset( $args['slug'] ) && $args['slug'] !== $old_status->slug )
-		) ) {
-			// translators: %s: Post status, like "Draft"
-			return new WP_Error( 'restricted', sprintf( __( 'Changing the name and slug of a restricted status (%s) is not allowed.', 'vip-workflow' ), $old_status->name ) );
+		// If the name was changed, we need to change the slug unless its banned from slug updates
+		if ( isset( $args['name'] ) && $args['name'] !== $old_status->name && ! self::is_status_banned_from_slug_changes( $old_status->slug ) ) {
```

Comment:
I think we can remove the `! self::is_status_banned_from_slug_changes( $old_status->slug )` check from here, as the code unsets `$args['slug']` if the status is banned below anyway.

Reply:
Tbh, I put it there for safety reasons just in case